library/ZendService/Twitter/Media.php

```diff
@@ -0,0 +1,251 @@
+        $httpClient->setParameterPost($payload);
+        $response = $httpClient->send();
+        return new Response($response);
+    }
+
+    /**
+     * Send the chunks of the file to Twitter
+     *
+     * @param Http\Client $httpClient
+     * @params array $params
+     * @return boolean
+     */
+    protected function appendUpload($httpClient, $params)
+    {
+        $payload = [
+            'command'  => 'APPEND',
+            'media_id' => $this->data['media_id'],
+        ];
+
+        set_error_handler($this->createErrorHandler(), E_WARNING);
+        $fileHandle = fopen($this->data['image_filename'], 'rb');
+        restore_error_handler();
+
+        if (! $fileHandle) {
+            throw new Exception\MediaUploadException('Failed to open the file in the APPEND method.');
+        }
+
+        while (! feof($fileHandle)) {
+            $data = fread($fileHandle, $this->chunkSize);
+
+            $payload['media_data'] = base64_encode($data);
+            $payload['segment_index'] = $this->data['segment_index']++;
+
+            $httpClient->resetParameters();
+            $httpClient->setHeaders(['Content-type' => 'application/x-www-form-urlencoded']);
+            $httpClient->setMethod('POST');
+            $httpClient->setParameterPost($payload);
+
+            $response = $httpClient->send();
+            $appendStatus = $response->isSuccess();
+
+            if (! $appendStatus) {
+                throw new Exception\MediaUploadException(
+                    'Failed uploading segment '
+                    . ($this->data['segment_index'] - 1)
+                    . ' of '
+                    . $this->data['image_filename']
+                    . '. Error Code: ' . $response->getStatusCode()
+                    . '. Reason: ' . $response->getReasonPhrase()
+                );
+            }
+        }
+
+        fclose($fileHandle);
+
+        return $appendStatus;
+    }
+
+    /**
+     * Send the upload finalize to Twitter
+     *
+     * @param Http\Client $httpClient
+     * @params array $params
+     * @return Twitter\Response
+     */
+    protected function finalizeUpload($httpClient, array $params)
+    {
+        $payload = [
+            'command'  => 'FINALIZE',
+            'media_id' => $this->data['media_id'],
+        ];
+
+        $httpClient->resetParameters();
+        $httpClient->setHeaders(['Content-type' => 'application/x-www-form-urlencoded']);
+        $httpClient->setMethod('POST');
+        $httpClient->setParameterPost($payload);
+        $response = $httpClient->send();
+
+        return new Response($response);
+    }
+
+    /**
+     * Validate that the file exists and can be opened.
+     *
+     * @todo Put a check to make sure the file is local.
+     * @param string $fileName
+     * @return boolean
+     */
+    protected function validateFile($fileName)
+    {
+        $returnValue = false;
+        $returnValue = file_exists($fileName);
+
+        if ($returnValue) {
```

Comment:
Looking through the docs, this can just be `is_readable()` (as it combines a `file_exists()` call at the C level). The operations need to be surrounded by an error handler, though, as they can emit an `E_WARNING`.

Updates coming momentarily.